cosim/black-parrot-example/ps.cpp

```diff
@@ -93,6 +120,39 @@ inline uint64_t get_counter_64(bp_zynq_pl *zpl, uint64_t addr) {
   } while (1);
 }
 
+unsigned clog2(unsigned val) {
+  unsigned i;
+  for(i = 0;i < 32;i++) {
+    if((1U << i) >= val)
+      break;
+  }
+  return i;
+}
+#define safe_clog2(x) (((x) == 1U) ? 1U : clog2(x))
+unsigned get_bsg_tag_reset_len(unsigned els, unsigned width) {
+  unsigned lg_width = clog2(width + 1);
+  unsigned header_len = safe_clog2(els) + 1 + lg_width;
+  unsigned max_packet_len = ((1 << lg_width) - 1 + header_len);
+  return (1 << (safe_clog2(max_packet_len + 1))) + 1;
+}
+
+void write_bsg_tag_packet(bp_zynq_pl *zpl, unsigned els, unsigned width,
```

Comment:
can we move this to include/common/bsg_tag_bitbang.h like we have for bsg_axil.h? That way it will act as a driver we can include in other examples